src/Agent.Listener/Environment.cs

```diff
@@ -0,0 +1,338 @@
+
+            GetFilteredEnvironmentVars(caps);
+
+            GetRegularCapabilities(caps, token);
+
+            await GetShellCapabilities(caps, token);
+
+            caps["Agent.Name"] = agentName ?? string.Empty;
+            //TODO: figure out what should be the value of Agent.OS
+            //XPLAT is using process.platform, which returns 'darwin', 'freebsd', 'linux', 'sunos' or 'win32'
+            //windows agent is printing environment variable "OS", which is something like "Windows_NT" even when running Windows 10
+            //.Net core API RuntimeInformation.OSDescription is returning "Microsoft Windows 10.0.10586", 
+            //"Linux 3.13.0-43-generic #72-Ubuntu SMP Mon Dec 8 19:35:06 UTC 2014", "Darwin 15.4.0 Darwin Kernel Version 15.4.0: Fri Feb 26 22:08:05 PST 2016;"
+            caps["Agent.OS"] = RuntimeInformation.OSDescription ?? string.Empty;
+            caps["Agent.ComputerName"] = System.Environment.MachineName ?? string.Empty;
+
+            foreach (var cap in caps)
+            {
+                Trace.Info($"Capability: {cap.Key} Value: {cap.Value}");
+            }
+
+            _capsCache = new Dictionary<string, string>(caps);
+
+            return caps;
+        }
+
+        private void GetRegularCapabilities(Dictionary<string, string> capabilities, CancellationToken token)
+        {
+            try
+            {
+                //TODO: allow paths to embed environment variables with "$VARNAME" or some other syntax and parse them
+                var paths = new string[]
+                    {
+                        Path.Combine(System.Environment.GetEnvironmentVariable("ANDROID_STUDIO") ?? string.Empty, "/tools/android"),
+                        Path.Combine(System.Environment.GetEnvironmentVariable("HOME") ?? string.Empty, "/Library/Developer/Xamarin/android-sdk-macosx/tools/android")
+                    };
+                _regularCapabilities.Add(new Capability("AndroidSDK", "android", paths));
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+            }
+
+            foreach (var cap in _regularCapabilities)
+            {
+                string capPath = IOUtil.Which(cap.Tool ?? cap.Name);
+                if (!string.IsNullOrEmpty(capPath))
+                {
+                    capabilities.Add(cap.Name, capPath);
+                }
+                else if (cap.Paths != null)
+                {
+                    foreach (var path in cap.Paths)
+                    {
+                        if (File.Exists(path))
+                        {
+                            capabilities.Add(cap.Name, path);
+                            break;
+                        }
+                    }
+                }
+
+                token.ThrowIfCancellationRequested();
+            }
+        }
+
+        private async Task GetShellCapabilities(Dictionary<string, string> capabilities, CancellationToken token)
```

Comment:
This name is misleading. In this agent, it's run via the process invoker (i.e. process). It looks like this name is hold-over from the xplat agent, where I'm guessing shelljs was used.